lib/agent.js

```diff
@@ -29,6 +29,19 @@ function Agent(backend, stream) {
   // Map from queryId -> emitter
   this.subscribedQueries = {};
 
+  // Track which documents are subscribed to presence by the client. This is a
+  // map of channel -> stream
+  this.subscribedPresences = {};
+  // Highest seq received for a subscription request. Any seq lower than this
+  // value is stale, and should be ignored. Used for keeping the subscription
+  // state in sync with the client's desired state
+  this.presenceSubscriptionSeq = 0;
```

Comment:
Have you been able to find or construct a scenario where this sequence guard is needed?

My understanding is that websockets and browserchannel both guarantee in-order delivery. I'm not sure about SockJS's XHR-based transports - some brief searching was inconclusive.

Even if the transport were naive XHR, meaning messages could get delivered out of order, I think this would need to be keyed by channel. A scenario - If a client sends `ps foo.a` then `ps foo.b`, but naive XHRs are received in the opposite order, then `ps foo.b` would set a new high agent-wide watermark, causing `ps foo.a` to get ignored.

Reply:
Ha yes, should definitely be per-channel. Good spot.

There's definitely a corner case where it's needed, even with in-order message guarantees:

1. Client subscribes
2. Client immediately, synchronously unsubscribes
3. Agent receives subscribe request, fires off an async request for a stream from the backend
4. Agent receives the unsubscribe request, which is processed synchronously (even if async, would be done in a different amount of time to the stream request)
5. Client is unsubscribed
6. Agent finally receives the subscription stream from backend, and subscribes
7. Client finishes subscribed, even though they requested the unsubscribe after the subscribe

This is obviously a super niche corner case, and I guess consumers shouldn't be _too_ surprised if they get weird behaviour when not waiting for their callbacks to resolve before making further requests, but I do think that this is the most "logical" thing to do.

If we think this is too niche for the maintenance overhead, I can just cut this behaviour.